### ISSUE TITLE
Use systemd socket activation for buildkitd ?

### ISSUE THREAD

**Issue by afbjorklund** [lima-vm/lima#423]
I added some patches to buildkit, to support systemd socket activation:

http://0pointer.de/blog/projects/socket-activation.html

This makes buildkitd start "on demand", so that the service can stay off...

Maybe this would be useful for lima `buildkitd`, too ?

**Comment by AkihiroSuda** [comment]
SGTM

**Comment by afbjorklund** [comment]
This is actually an issue from nerdctl (full), not from the lima code:

```Dockerfile
# NOTE: github.com/moby/buildkit/examples/systemd is not included in BuildKit v0.8.x, will be included in v0.9.x
RUN cd /out/lib/systemd/system && \
  sedcomm='s@bin/containerd@bin/buildkitd@g; s@(Description|Documentation)=.*@@' && \
  sed -E "${sedcomm}" containerd.service > buildkit.service && \
  echo "" >> buildkit.service && \
  echo "# This file was converted from containerd.service, with \`sed -E '${sedcomm}'\`" >> buildkit.service
```

The upstream files are available here: https://github.com/moby/buildkit

```
examples/systemd/system/buildkit.service
examples/systemd/system/buildkit.socket
examples/systemd/user/buildkit-proxy.service
examples/systemd/user/buildkit-proxy.socket
examples/systemd/user/buildkit.service
```

---

But because of the complexity of having to forward `fd://3`, probably not worth it?

https://www.freedesktop.org/software/systemd/man/latest/systemd-socket-proxyd.html
